Add more minor typehints

mathiasertl opened the pull request:
Hi,

I hereby  add more minor typehints that I've been missing in my own code that does strict mypy checks.

I'm open to contributing more typehints in this project, but I'm not sure it is wanted? (some projects don't want full typehinted consistency everywhere).

kr, Mat

mathiasertl commented on dns/resolver.py:
```diff
@@ -1293,7 +1293,7 @@ def resolve_address(self, ipaddr: str, *args: Any, **kwargs: Any) -> Answer:
         return self.resolve(
             dns.reversename.from_address(ipaddr), *args, **modified_kwargs
-        )  # type: ignore[arg-type]
+        )
```
Note: Mypy says this is an unused override.

rthalley commented:
Type hinting in the public API is a good thing, so contributions that fix things you've noticed calling into dnspython are good.  Note that dnspython's continuous integration runs a type check, and the settings it uses for mypy include "--disallow-incomplete-defs" so if you add an annotation to any part of a function signature, you have to do everything.  That's why the tests are all failing for your PR so far, as e.g. you added -> None as a return type but didn't give types for parameters.  Look at the "potype" rule in the Makefile and mypy.ini to see exactly how dnspython checks.

mathiasertl commented on dns/resolver.py:
```diff
@@ -447,13 +447,13 @@ def __init__(self, key, value):
         self.prev = self
```
For my next MR:

This class is currently a bit problematic. The "sentinel" object is instantiated as `LRUCacheNode(None, None)`, implying that the arguments are optional. But then `get()` etc of the cache thing that key/value might be None, throwing lots of type errors.

A solution would be to introduce a `SentinelCacheNode` class that allows None values, or to `# type: ignore` the instantiation of the sentinel node.

What do you think?

rthalley replied:
I don't think I'd change this, as the LRUCacheNode is really internal to the implementation and is not exposed.  Maybe just an assert that node.value is not None if needed?

rthalley replied:
Hmm, I just reread and see I missed part of your point.  I'll ponder this a bit more.  I'm also ok with just leaving this class untyped as it's basically internal.  (In retrospect, I should have called it _LRUCacheNode).  But I'll think about your idea a little more too.

mathiasertl commented:
Hi @rthalley ,

I fixed the issues and removed the `-> None` where not trivially fixed.

Now I have a python 3.8 failure in pytest - this looks like a transient issue, but I cannot retrigger the job. Can you retry?

kr, Mat

rthalley commented:
The 3.8 build issue was definitely a transient.  I haven't forgotten your PR, but won't be looking at it until the weekend.

/Bob
